sources/packages/backend/apps/api/src/services/student/student.service.ts

```diff
@@ -272,6 +275,32 @@ export class StudentService extends RecordDataModelService<Student> {
       } as StudentAccountApplication;
       await entityManager.getRepository(StudentUser).save(studentUser);
 
+      // Add hold restriction if a partial match was found and the account is not bridge matched
+      if (shouldAddHoldRestriction && !sfasIndividual && !existingStudent) {
+        const holdNote = await this.noteSharedService.createStudentNote(
+          savedStudent.id,
+          NoteType.Restriction,
+          "Restriction added to prevent application completion while potential partial match exists",
+          this.systemUsersService.systemUser.id,
+          entityManager,
+        );
+
+        const holdRestriction = new StudentRestriction();
+        holdRestriction.student = { id: savedStudent.id } as Student;
+        holdRestriction.restriction = await entityManager
```

Comment:
Can we check for an existing active hold restriction before adding it? similar to this. 

<img width="912" height="508" alt="image" src="https://github.com/user-attachments/assets/5f596f6b-1d66-40ac-ac36-11e1fade01f1" />

There is a very edge case possibility that, when student switch from basic-bceid to bcsc or vice-versa then the student profile creation can happen more than once for a same student which will introduce the possibility of them already having a HOLD restriction.